cpp/json_ffi/conv_template.cc

```diff
@@ -0,0 +1,310 @@
+#include "conv_template.h"
+#include "../metadata/json_parser.h"
+
+using namespace mlc::llm;
+using namespace mlc::llm::json_ffi;
+
+
+std::map<MessagePlaceholders, std::string> PLACEHOLDERS = {
+    {MessagePlaceholders::SYSTEM, "{system_message}"},
+    {MessagePlaceholders::USER, "{user_message}"},
+    {MessagePlaceholders::ASSISTANT, "{assistant_message}"},
+    {MessagePlaceholders::TOOL, "{tool_message}"},
+    {MessagePlaceholders::FUNCTION, "{function_string}"}
+};
+
+MessagePlaceholders message_placeholder_from_string(const std::string& role) {
+    static const std::unordered_map<std::string, MessagePlaceholders> enumMap = {
```

Comment:
Snake case for variables. See google c++ style guide https://google.github.io/styleguide/cppguide.html